phpBB/phpbb/extension/extension_interface.php

```diff
@@ -27,6 +27,13 @@
 	public function is_enableable();
 
 	/**
+	* Indicate why the extension can not be enabled.
+	*
+	* @return string|array with actual processed language string(s) (not language keys), or empty
+	*/
+	public function not_enableable_reason();
```

Comment:
Just for the processing of the language strings, ok.  Otherwise, identical to my proposal.
But again, for that, I would add a method to the language object that would take the "complex" structure and get the clean array of translated messages out (using code similar to the one proposed by Nicofuma).  But, in my view, this is not part of this PR.